README.md

```diff
@@ -1 +1,107 @@
+
+### Vehicle registration
+
+`POST /vehicles`
+
+Request body:
+
+```json
+{ "id": "some-uuid-here" }
+```
+
+Response status code: 204
+
+Response body should be empty.
+
+The vehicle will contact this endpoint to register itself in your solution. It will pass in a universally unique ID to identify itself. After receiving this registration request, you should expect to start receiving vehicle location updates from the vehicle.
+
+**Example:** the vehicle with the ID `abc123` wants to start sending location updates. It would contact `POST /vehicles` with ``{ "id": “abc123” }`` in the request body.
+
+### Location update
+
+`POST /vehicles/:id/locations`
+
+Request body:
+
+```json
+{ "lat": 10.0, "lng": 20.0, "at": "2017-09-01T12:00:00Z" }
+```
+
+Response status code: 204
+
+Response body should be empty.
+
+The vehicle will contact this endpoint whenever a new location is detected, but no more than once every 3 seconds. It will contain the latitude and longitude of its position, and the timestamp that the position was recorded in ISO8601 format. The vehicle will never contact this endpoint without having registered first.
+
+**Example:** the vehicle with the ID `abc123` has recorded a new position at 12pm CET on 2nd December 2017, at the position (51.2, 45.3). It would contact `POST /vehicles/abc123/locations` with ``{ "lat": 51.2, “lng”: 45.3, “at”: “2017-12-02T12:00:00+01:00” }`` in the request body.
+
+### Vehicle de-registration
+
+`DELETE /vehicles/:id`
+
+Response status code: 204
+
+Response body should be empty.
+
+The vehicle will contact this endpoint when it will stop emitting location updates. After this endpoint has been called, the vehicle will never send further location updates unless it calls the vehicle registration endpoint again.
+
+**Example:** the vehicle with the ID `abc123` has finished emitting location updates. It would contact `DELETE /vehicles/abc123` to indicate this.
+
+## Visualizing vehicle location
+
+You are free to choose how to visualize the current vehicle positions, but the most obvious choice would be to display them on a map in some way. You must also visualize the bearing/direction of the vehicle.
+
+You can look at the Google Maps, Mapbox and Leaflet libraries as options to help you with building a map visualization. Try to consider how easy your visualization is to understand. Some concepts which can help to improve the visualization:
+
+* Group clusters of vehicles together
+* Modify your visualization depending on zoom level
+
+We expect your visualization to be accessible via a web browser. It only needs to be tested against the most recent version of Chrome.
+
+## Additional business requirements
+
+If a vehicle has exited the "city boundaries", the system should disregard any GPS emissions from that vehicle. The city boundary is currently assumed to be a 50km radius around door2door's office which is located at latitude 52.53 and longitude 13.403.
```

Comment:
We should make sure the simulator emits some GPS coordinates beyond the boundaries (not a blocker on this PR, just a comment).